invenio_oauthclient/utils.py

```diff
@@ -205,10 +286,11 @@ def fill_form(form, data):
     """
     for (key, value) in data.items():
         if hasattr(form, key):
+            field = getattr(form, key)
             if isinstance(value, dict):
-                fill_form(getattr(form, key), value)
-            else:
-                getattr(form, key).data = value
+                fill_form(field, value)
+            elif field is not None:
+                field.data = value
```

Comment:
avoiding setting the `data` attribute for `None` fields allows us to properly disable "prominent" fields (e.g. `email`), instead of just hiding them